sql/catalyst/src/main/scala/org/apache/spark/sql/connector/write/LogicalWriteInfoImpl.scala

```diff
@@ -23,4 +23,6 @@ import org.apache.spark.sql.util.CaseInsensitiveStringMap
 private[sql] case class LogicalWriteInfoImpl(
     queryId: String,
     schema: StructType,
-    options: CaseInsensitiveStringMap) extends LogicalWriteInfo
+    options: CaseInsensitiveStringMap,
+    override val rowIdSchema: StructType = null,
+    override val metadataSchema: StructType = null) extends LogicalWriteInfo
```

Comment:
shall we make these two parameters optional as not every write needs to specify them?

Reply:
Will do.

Reply:
Could you address the review comments?

Reply:
Done.